types/waterline/index.d.ts

```diff
@@ -12,21 +12,37 @@ declare namespace Waterline {
     type Connection = {
         adapter: string;
     }
+    type ConnectionV3 = {
```

Comment:
I couldn't tell from the documentation, but is it necessary to have definitions for the existing and the "V3" versions? Would it not make more sense to just use the updated definitions where there is sufficient overlap with the existing types? For example, `ConnectionV3` is assignment-compatible with `Connection` because all of the additional properties are optional.

Reply:
Well v2 and v3 are not compatible at all, i can use the existing definition but my point was to note the differences.  i will update the existing types, keep in mind the type array is valid in version 2 but not in version 3.  

The actual version of waterline was build to make ease migrate from different database vendor, so the types for define tables are reduced to most compatible "string", "number", "boolean", "json" and "ref" plus definition to make 1-n and n-m relations

Reply:
I update the existing types.